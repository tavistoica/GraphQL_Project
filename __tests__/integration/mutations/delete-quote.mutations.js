const { createTestClient } = require("apollo-server-testing");
const { gql } = require("apollo-server");
const {
  database: { url },
} = require("../../../src/core/config/test.config");
const Quote = require("../../../src/entities/quote/quote.entity");
const { generateID } = require("../../../src/core/helpers/generate-id.helper");
const { Server } = require("../../../src/bootstrap/server.bootstrap");
const { Database } = require("../../../src/bootstrap/database.bootstrap");

const server = new Server();
const database = new Database();

const quoteID = "123";

function doMutation() {
  return gql`
    mutation DeleteQuotes {
      deleteQuote(id: quoteID) {
        id
      }
    }
  `;
}

function getQuery() {
  return gql`
    query getQuotes {
      quotes {
        quoter
        phrase
      }
    }
  `;
}

function createQuote(quote) {
  const newQuote = new Quote(quote);
  return newQuote.save();
}

async function createContext() {
  const quote = await createQuote({
    id: quoteID,
    phrase: "O vorba mare",
    quoter: "mafiotu",
  });
  return {
    doQuotesMutation: doMutation(),
    getQuotesQuery: getQuery(),
    quote,
  };
}

beforeEach(async () => {
  await database.bootstrap({ url });
  server.bootstrap();
});
afterEach(() => database.reset());

it("deleteQuote Mutation use case Test", async () => {
  const { doQuotesMutation, getQuotesQuery } = await createContext();
  const { query, mutate } = createTestClient(server.apolloClient);
  await mutate({
    mutation: doQuotesMutation,
  });
  const res = await query({
    query: getQuotesQuery,
  });
  expect(res).toMatchSnapshot();
});
