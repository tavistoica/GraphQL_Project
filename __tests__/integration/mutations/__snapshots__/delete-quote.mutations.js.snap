// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteQuote Mutation use case Test 1`] = `
Object {
  "data": Object {
    "quotes": Array [
      Object {
        "phrase": "O vorba mare",
        "quoter": "mafiotu",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
